ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/TransactionReceipt.java

```diff
@@ -137,15 +168,22 @@ public static TransactionReceipt readFrom(final RLPInput input) {
       // TODO consider validating that the logs and bloom filter match.
       final LogsBloomFilter bloomFilter = LogsBloomFilter.readFrom(input);
       final List<Log> logs = input.readList(Log::readFrom);
+      final Optional<String> revertReason;
+      if (input.isEndOfCurrentList()) {
+        revertReason = Optional.empty();
+      } else {
+        final byte[] bytes = input.readBytesValue().getArrayUnsafe();
+        revertReason = Optional.of(new String(bytes, StandardCharsets.UTF_8));
```

Comment:
We need a separate method which checks if a revert reason was available and parses it.  When we receive receipts over the wire, it's a protocol violation if they have the extra revert reason included and should be rejected.

Reply:
Could that be done at the level above this? i.e. a TransactionReceipt having or not having a revert reason is ok - however if it comes via a message _then_ it is illegal ... could the message receiver/handler raise the exception if the TR in the message contains a Revert?

Reply:
As long as we do reject the message it's probably ok.  Seems cleaner to me to do it here - ultimately just by passing in a boolean to say if the revert reason is allowed.  But I'm not overly fussed.

Reply:
Done